### PR TITLE
Feat: add async validation on uploading file

### PR DESCRIPTION
**Pull request by ArthurTriis1**
#### What problem is this solving?

When the first version of Bulk Import was launched, the B2B team quickly realized the need for asynchronous file validation. To achieve this, the route responsible for uploading and validating the file was split into two parts: the first for uploading the file and the second for initiating its validation. This PR integrates b2b-organizations with the new validation routes. The following sequence occurs after the user uploads a new file:

1. The Uploading route is called with the file.
2. The Validation route is called, and the modal transitions to the "awaiting validation" state.
3. The Item Details route is called every 5 seconds, checking the status. Upon successful validation, the modal can be closed, and the import process begins. In case of validation issues, the modal enters a reporting state, allowing the user to view more details about the error from the validation file.

During each request, if an unexpected error occurs, the modal enters a generic error state, allowing the user to restart the process.

#### How to test it?

[Workspace](https://validating--b2bstoreqa.myvtex.com/admin/b2b-organizations/organizations#/organizations)

#### Screenshots or example usage:

Screenshot 2024-01-24 at 18 28 37

#### How does this PR make you feel? [:link:](http://giphy.com/)

[B2BTEAM-1544](https://vtex-dev.atlassian.net/browse/B2BTEAM-1544)

[B2BTEAM-1544]: https://vtex-dev.atlassian.net/browse/B2BTEAM-1544?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: react/components/utils/bytesToSize.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const bytesToSize = (bytes: number) => {
+  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB']
+
+  if (bytes === 0) return '0 Bytes'
```
Comment (mairatma) [inline react/components/utils/bytesToSize.ts:4]: Should this be translated? In Portuguese I think we use the same term, but not sure if we can guarantee the same for other languages.

File: react/hooks/useBulkImportDetailsQuery.ts [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
+  onSuccess,
+  refreshInterval = 0,
+}: UseBulkImportDetailsQueryProps) => {
   return useSWR(
     importId ? `/buyer-orgs/${importId}` : null,
     () => getBulkImportDetails(importId),
     {
+      refreshInterval,
       revalidateOnFocus: false,
+      onSuccess: data => {
```
Comment (mairatma) [inline react/hooks/useBulkImportDetailsQuery.ts:23]: I think that you can just pass `onSuccess` here:

```suggestion
      onSuccess,
```

File: react/hooks/useValidateBulkImport.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+const useValidateBulkImport = ({ onSuccess }: { onSuccess?: () => void }) => {
+  const { trigger } = useSWRMutation(
+    '/buyer-orgs/start',
+    (_, { arg }: { arg: { importId: string } }) =>
+      validateBulkImport(arg.importId),
+    {
+      onSuccess() {
+        onSuccess?.()
+      },
```
Comment (mairatma) [inline react/hooks/useValidateBulkImport.ts:13]: Can this just be:

```suggestion
      onSuccess,
```

File: react/components/utils/bytesToSize.ts (hunk 1 above)
Comment (ArthurTriis1) [reply react/components/utils/bytesToSize.ts:4]: Oh, good question! I will ask to translation team and confirm.

**Review by mairatma — APPROVED** [review]

**Pull request merged**